File: apps/backend/app/schemas/demo_scenario.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class DemoScenarioRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    scenario_key: str
    title: str
    description: str | None = None
    event_source: str
    event_type: str
    market_event: dict[str, Any]
    research_evidence: list[dict[str, Any]] = Field(default_factory=list)
    expected_recommendation: str | None = None
    default_impact_score: float
    default_complexity_score: float
    default_urgency_score: float
    tags: list[str] = Field(default_factory=list)
    is_active: bool
    created_at: datetime
    updated_at: datetime


class DemoTriggerRequest(BaseModel):
    company_profile_id: UUID | None = None
    demo_speed: str = "normal"

File: apps/frontend/src/types/demo-scenario.ts
export interface DemoScenario {
  id: string;
  scenario_key: string;
  title: string;
  description?: string | null;
  event_source: string;
  event_type: string;
  market_event: {
    title?: string;
    summary?: string;
    why_it_matters?: string;
    recommended_evolution?: string;
    importance_score?: number;
    [key: string]: unknown;
  };
  research_evidence: Array<Record<string, unknown>>;
  expected_recommendation?: string | null;
  default_impact_score: number;
  default_complexity_score: number;
  default_urgency_score: number;
  tags: string[];
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export type DemoSpeed = "fast" | "normal" | "slow";
